File: java/src/main/java/io/fivexer/sdk/model/WorkerSkillSet.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A worker's skills after replacing the whole set.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerSkillSet {
    private String workerId;
    private List<WorkerSkill> skills;

    public String getWorkerId() { return workerId; }
    public List<WorkerSkill> getSkills() { return skills; }
}

File: php/src/Model/WorkerSkillSet.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A worker's skills after replacing the whole set.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerSkillSet
{
    public function __construct(
        public readonly string $workerId,
        /** @var list<WorkerSkill> */
        public readonly array $skills,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            skills: Json::parseEach($data, 'skills', [WorkerSkill::class, 'fromArray']),
        );
    }
}
